File: 003.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        vector<int> c(256,-1); 
        int b = 0, ans = 0; 
        if (s.size() == 1) return 1; 
        if (s.size() == 0) return 0; 
        c[s[0]] = 0;
        for (int e = 1 ; e < s.size(); e++){
            if (c[s[e]] != -1 ) {
                ans = e-b > ans? e-b: ans; 
                do {
                    c[s[b]] = -1; 
                    b++; 
                }while(b <= c[s[e]]);
                c[s[e]] = e;
            }
            else if (e == s.size()-1) ans = e-b+1 > ans? e-b+1: ans; 
            c[s[e]] = e;   
        }
        return ans; 
    }
};

File: 003.py
class Solution:
    
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0 
        
        self.storage = {}
        c1 = 0

        maxStringSize = 0 
        
        for c2, letter in enumerate(s):
            self.addToStorage(letter) 
            maxStringSize = max(maxStringSize, c2 - c1)     

            while c1 < c2 and  self.isRepeated(letter): 
                self.removeFromStorage(s[c1])
                c1 += 1 

        maxStringSize = max(maxStringSize, c2- c1 + 1)
        return maxStringSize 
    
    
    def addToStorage(self, letter): 
        if letter not in self.storage.keys():
            self.storage[letter] = 1
        else:
            self.storage[letter] += 1
    
    def removeFromStorage(self, letter):
        if letter in self.storage.keys():
            self.storage[letter] -= 1
    
    
    # Return if the letter is repeated inside the storage.
    def isRepeated(self, letter): 
        if letter in self.storage.keys():
            return self.storage[letter] > 1 
        
        return False 
    
